File: examples/v2/schedules/python/__main__.py
import pulumi
import pulumi_pulumiservice.v2 as ps_v2

config = pulumi.Config()
service_org = config.get("serviceOrg") or "service-provider-test-org"
project_name = config.get("projectName") or "pulumi-service-schedules-example"
stack_name = config.get("stackName") or "dev"
schedule_cron = config.get("scheduleCron") or "0 7 * * *"

parent_stack = ps_v2.Stack(
    "parentStack",
    org_name=service_org,
    project_name=project_name,
    stack_name=stack_name,
)

parent_settings = ps_v2.DeploymentSettings(
    "parentSettings",
    org_name=service_org,
    project_name=project_name,
    stack_name=stack_name,
    source_context={"git": {"repoUrl": "https://github.com/example/example.git", "branch": "refs/heads/main"}},
    opts=pulumi.ResourceOptions(depends_on=[parent_stack]),
)

nightly_deploy = ps_v2.ScheduledDeployment(
    "nightlyDeploy",
    org_name=service_org,
    project_name=project_name,
    stack_name=stack_name,
    schedule_cron=schedule_cron,
    request={"operation": "update"},
    opts=pulumi.ResourceOptions(depends_on=[parent_settings]),
)

pulumi.export("nightlyCron", nightly_deploy.schedule_cron)

File: examples/v2/schedules/go/main.go
package main

import (
	stacks "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/stacks"
	deployments "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v2/deployments"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		serviceOrg := cfg.Get("serviceOrg")
		if serviceOrg == "" {
			serviceOrg = "service-provider-test-org"
		}
		projectName := cfg.Get("projectName")
		if projectName == "" {
			projectName = "pulumi-service-schedules-example"
		}
		stackName := cfg.Get("stackName")
		if stackName == "" {
			stackName = "dev"
		}
		scheduleCron := cfg.Get("scheduleCron")
		if scheduleCron == "" {
			scheduleCron = "0 7 * * *"
		}

		parentStack, err := stacks.NewStack(ctx, "parentStack", &stacks.StackArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: pulumi.String(projectName),
			StackName:   pulumi.String(stackName),
		})
		if err != nil {
			return err
		}

		parentSettings, err := deployments.NewSettings(ctx, "parentSettings", &deployments.SettingsArgs{
			OrgName:     pulumi.String(serviceOrg),
			ProjectName: pulumi.String(projectName),
			StackName:   pulumi.String(stackName),
			SourceContext: pulumi.Map{
				"git": pulumi.Map{
					"repoUrl": pulumi.String("https://github.com/example/example.git"),
					"branch":  pulumi.String("refs/heads/main"),
				},
			},
		}, pulumi.DependsOn([]pulumi.Resource{parentStack}))
		if err != nil {
			return err
		}

		nightlyDeploy, err := deployments.NewScheduledDeployment(ctx, "nightlyDeploy", &deployments.ScheduledDeploymentArgs{
			OrgName:      pulumi.String(serviceOrg),
			ProjectName:  pulumi.String(projectName),
			StackName:    pulumi.String(stackName),
			ScheduleCron: pulumi.String(scheduleCron),
			Request: pulumi.Map{
				"operation": pulumi.String("update"),
			},
		}, pulumi.DependsOn([]pulumi.Resource{parentSettings}))
		if err != nil {
			return err
		}

		ctx.Export("nightlyCron", nightlyDeploy.ScheduleCron)
		return nil
	})
}
